java/src/main/java/com/google/appengine/tools/mapreduce/impl/shardedjob/ShardedJobRunner.java

```diff
@@ -333,27 +327,34 @@ private void handleLockHeld(String taskId, ShardedJobStateImpl<T> jobState,
             "Resuming after abandon lock for " + taskId + " on slice: "
                 + taskState.getSequenceNumber()), true);
       } else {
-        retryState = handleShardFailure(jobState, taskState, new RuntimeException(
-          "Lock for " + taskId + " expired on slice: " + taskState.getSequenceNumber()));
+        retryState = handleSliceFailure(jobState, taskState, new RuntimeException(
```

Comment:
watch out, changing handle**Shard**Failure by handle**Slice**Failure

Reply:
yeah, intentional; I don't really know why it went straight to failing the `shard` in the `else` case before; but certianly that's not what we actually want to do (in case the cause of slice dying was something transient, in which case we *should* retry slice if allowed; if not allowed, it becomes a shard failure if you look at the implementation)